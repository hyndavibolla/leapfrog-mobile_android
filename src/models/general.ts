import { Dispatch } from 'react';
import Auth0 from 'react-native-auth0';

import { ApiService } from '_services/ApiService';
import { AppService } from '_services/AppService';
import { CCPAService } from '_services/CCPAService';
import { ICrashlyticsService } from '_services/CrashlyticsService';
import { EventTrackerService } from '_services/EventTrackerService';
import { IHttpService } from '_services/HttpService';
import { NativeHelperService } from '_services/NativeHelperService';
import { Logger } from '_services/Logger';
import { IPerformanceTrackingService } from '_services/PerformanceTrackingService';
import { RaiseService } from '_services/RaiseService';
import { RemoteConfigService } from '_services/RemoteConfigService';
import { ICardLinkState } from '_state_mgmt/cardLink/state';
import { ICoreState } from '_state_mgmt/core/state';
import { IGameState } from '_state_mgmt/game/state';
import { IGiftCardState } from '_state_mgmt/giftCard/state';
import { IMessagesState } from '_state_mgmt/messages/state';
import { IMissionState } from '_state_mgmt/mission/state';
import { IRewardState } from '_state_mgmt/reward/state';
import { ISurveyState } from '_state_mgmt/survey/state';
import { IUserState } from '_state_mgmt/user/state';

export interface IAction<T = string, P = any> {
  type: T;
  payload: P;
}

export type IReducer<S = any, A = any> = (state: S, action: IAction<A>) => S;

export interface Deps {
  apiService: ApiService;
  appService: AppService;
  crashlyticsService: ICrashlyticsService;
  httpService: IHttpService;
  performanceTrackingService: IPerformanceTrackingService;
  raiseService: RaiseService;
  nativeHelperService: NativeHelperService;
  stateSnapshot: { get: () => IGlobalState; set: (state: IGlobalState) => void };
  logger: Logger;
  eventTrackerService: EventTrackerService;
  ccpaService: CCPAService;
  remoteConfigService: RemoteConfigService;
  authService: Auth0;
}
export type IGlobalState = {
  core: ICoreState;
  user: IUserState;
  game: IGameState;
  mission: IMissionState;
  reward: IRewardState;
  survey: ISurveyState;
  messages: IMessagesState;
  cardLink: ICardLinkState;
  giftCard: IGiftCardState;
};

export interface IGlobalContext {
  state: IGlobalState;
  dispatch: Dispatch<IAction<any, any>>;
  deps: Deps;
}

export enum HTTP_METHOD {
  GET = 'GET',
  POST = 'POST',
  PUT = 'PUT',
  PATCH = 'PATCH',
  DELETE = 'DELETE'
}

export interface ITokenSet {
  accessToken: string;
  refreshToken: string;
  accessTokenExpiryTime: string;
  sywToken?: string;
  /**
   * @deprecated deviceId does not belong to ITokenSet anymore.
   */
  deviceId?: string;
  scope?: string;
  expiresIn?: number;
  tokenType?: string;
}

export enum FeatureFlag {
  API_OVERRIDE = 'API_OVERRIDE',
  TEST_ONLY = 'TEST_ONLY',
  ENV_BANNER = 'ENV_BANNER',
  BUTTON_DEBUG = 'BUTTON_DEBUG',
  LOG_LEVEL_INFO = 'LOG_LEVEL_INFO',
  LOG_LEVEL_DEBUG = 'LOG_LEVEL_DEBUG',
  LOG_LEVEL_WARN = 'LOG_LEVEL_WARN',
  LOG_LEVEL_ERROR = 'LOG_LEVEL_ERROR',
  LOG_VIEW = 'LOG_VIEW',
  STREAK_REDEEM_CARD = 'STREAK_REDEEM_CARD',
  SAILTHRU_TRACKING = 'SAILTHRU_TRACKING',
  SURVEY = 'SURVEY',
  PREDICTIVE_MAP_SEARCH = 'PREDICTIVE_MAP_SEARCH',
  MISSIONS_SEARCH = 'MISSIONS_SEARCH',
  SEE_ALL_TRANSACTIONS = 'SEE_ALL_TRANSACTIONS',
  WALLET_YOUR_GC = 'WALLET_YOUR_GC'
}

export type DateLike = Date | string | number;

export type HTMLString = string;

export interface IGameLevelUI {
  updatedAt: DateLike;
  level: number;
}

export interface ICCPASetting {
  allow: boolean;
}

export interface IOnboarding {
  seenOnboarding: boolean;
}

export enum ButtonCreativeType {
  HERO = 'hero',
  CAROUSEL = 'carousel',
  LIST = 'list',
  GRID = 'grid',
  DETAIL = 'detail',
  OTHER = 'other'
}

export interface ISearchHistoryItem {
  keyword: string;
  lastUpdatedAt: DateLike;
}

export enum TooltipKey {
  EARN = 'earn',
  REWARDS = 'rewards',
  MISSIONS = 'missions',
  STREAK = 'streak'
}

export interface IApiOverrideRule {
  key: string;
  description?: string;
  matcher: string; // reg exp
  status: number;
  response: any;
}

export namespace RemoteConfigModel {
  export interface IAppVersion {
    suggested: number;
    required: number;
  }

  export interface IEarnSection {
    name: string;
    order: number;
    visible: boolean;
    title: string;
    description: string;
    seeAllButton: boolean;
  }
}

export enum RUNTIME_ENV {
  LIVE = 'live',
  DEV_TEST = 'test'
}

export interface IMeasure {
  x: number;
  y: number;
  width: number;
  height: number;
}

export interface IExpirePointsBannerDataSet {
  lastPointsDate: DateLike;
  lastAcceptedDate: DateLike;
}

export interface ISailthruMessage {
  createdAt: string;
  custom: {
    [k: string]: string;
  };
  htmlText: string;
  id: string;
  isRead: boolean;
  text: string;
  title: string;
  type: string;
  cardImageUrl?: string;
  share: boolean;
}

export interface ITutorialStatus {
  isBannerWatched: boolean;
  date: string;
}
