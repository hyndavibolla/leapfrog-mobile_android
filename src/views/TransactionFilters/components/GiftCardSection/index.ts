export { default as GiftCardSection } from './GiftCardSection';
