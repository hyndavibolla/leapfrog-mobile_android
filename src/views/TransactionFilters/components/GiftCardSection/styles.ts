import { css } from '@emotion/native';
import { COLOR, FONT_FAMILY, FONT_SIZE } from '_constants';

export const styles = {
  note: css`
    color: ${COLOR.BLACK};
    font-family: ${FONT_FAMILY.BOLD};
    font-size: ${FONT_SIZE.SMALLER};
    line-height: ${FONT_SIZE.BIG};
    margin-bottom: 16px;
  `,
  bubbles: css`
    background-color: ${COLOR.WHITE};
    border-radius: 8px;
    flex-direction: row;
    margin-horizontal: 0;
  `,
  specialBubbles: css`
    padding-horizontal: 24px;
  `,
  bubblesContainer: css`
    flex: 1;
    justify-content: space-around;
  `,
  bubble: css`
    margin-right: 16px;
  `,
  SpecialBubble: css`
    margin-right: 0;
  `,
  container: css`
    padding: 24px 8px 24px 24px;
  `
};
