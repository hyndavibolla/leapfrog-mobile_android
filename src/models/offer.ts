import { DateLike } from './general';

export enum PointsType {
  EARN = 'earn',
  REDEEM = 'redeem',
  SURPRISE_REDEEM = 'surpriseRedeem',
  POINTS_EXPIRY = 'pointsExpiry',
  POINTS_AVAILABLE = 'pointsAvailable'
}

export enum ProgramSubCategory {
  GAS = 'gas',
  GROCERY = 'grocery',
  RESTAURANT = 'restaurant',
  SEARS = 'sears',
  KMART = 'kmart',
  OTHER = 'other',
  SURVEY = 'survey',
  MISSION = 'mission'
}

export enum ProgramType {
  SPEED = 'speed',
  BIG = 'big',
  STREAK = 'SYWMAX-Streak',
  ONBOARDING = 'onboarding',
  SYWMAX = 'SYWMAX',
  CARDLINK = 'cardlink',
  CITI = 'CITI'
}

export enum TransactionFilter {
  ALL_TRANSACTIONS = 'all-transactions',
  LOCAL_OFFERS = 'local-offers',
  MISSIONS = 'missions',
  REWARDS = 'rewards',
  SYW_MASTERCARD = 'SYW-mastercard'
}

export interface IOffer {
  id: string;
  points: number;
  pointsType: PointsType;
  programSubCategory: ProgramSubCategory;
  programType: ProgramType;
  name?: string;
  pointStartDate: DateLike;
  pointEndDate: DateLike;
  dollarRedeemed: number;
  rewardOfferCode?: string;
  isStreak?: boolean;
}
