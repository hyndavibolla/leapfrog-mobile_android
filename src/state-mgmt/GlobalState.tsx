import React, { createContext, memo, PropsWithChildren, useReducer, Dispatch, useEffect } from 'react';

import { IGlobalState, Deps, IReducer, IGlobalContext } from '_models/general';
import { reducer as coreReducer } from './core/reducer';
import { reducer as userReducer } from './user/reducer';
import { reducer as gameReducer } from './game/reducer';
import { reducer as missionReducer } from './mission/reducer';
import { reducer as rewardReducer } from './reward/reducer';
import { reducer as surveyReducer } from './survey/reducer';
import { reducer as messagesReducer } from './messages/reducer';
import { reducer as cardLinkReducer } from './cardLink/reducer';
import { reducer as giftCardReducer } from './giftCard/reducer';
import { initialState as coreInitialState } from './core/state';
import { initialState as userInitialState } from './user/state';
import { initialState as gameInitialState } from './game/state';
import { initialState as missionInitialState } from './mission/state';
import { initialState as rewardInitialState } from './reward/state';
import { initialState as surveyInitialState } from './survey/state';
import { initialState as messagesInitialState } from './messages/state';
import { initialState as cardLinkInitialState } from './cardLink/state';
import { initialState as giftCardInitialState } from './giftCard/state';
import { combineReducers } from '_utils/combineReducers';
import { GeneralModel } from '_models';
import { deepClone } from '_utils/deepClone';

export const getInitialState = () =>
  deepClone<IGlobalState>({
    core: coreInitialState,
    user: userInitialState,
    game: gameInitialState,
    mission: missionInitialState,
    reward: rewardInitialState,
    survey: surveyInitialState,
    messages: messagesInitialState,
    cardLink: cardLinkInitialState,
    giftCard: giftCardInitialState
  });

export const combinedReducer = combineReducers<GeneralModel.IGlobalState>({
  core: coreReducer,
  user: userReducer,
  game: gameReducer,
  mission: missionReducer,
  reward: rewardReducer,
  survey: surveyReducer,
  messages: messagesReducer,
  cardLink: cardLinkReducer,
  giftCard: giftCardReducer
});

export const GlobalContext = createContext<{ state: IGlobalState; dispatch: Dispatch<any>; deps: Deps }>({
  state: getInitialState(),
  dispatch: /* istanbul ignore next */ () => null,
  deps: null
} as IGlobalContext);

export const GlobalProvider = memo(
  ({
    children,
    deps,
    initState,
    combinedReducers
  }: PropsWithChildren<{
    deps: Deps;
    initState: IGlobalState;
    combinedReducers: IReducer<IGlobalState>;
  }>) => {
    const [state, dispatch] = useReducer(combinedReducers, initState);
    useEffect(() => deps.stateSnapshot.set(state), [state]); // eslint-disable-line react-hooks/exhaustive-deps
    return <GlobalContext.Provider value={{ state, dispatch, deps } as IGlobalContext}>{children}</GlobalContext.Provider>;
  }
);
