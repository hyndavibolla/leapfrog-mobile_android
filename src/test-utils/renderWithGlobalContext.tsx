import React, { ReactElement } from 'react';
import { render } from '@testing-library/react-native';

import { getInitialState, GlobalProvider } from '../state-mgmt/GlobalState';
import { Deps, IGlobalState, IReducer } from '../models/general';
import { getMockDeps } from './getMockDeps';
import { getMockReducer } from './getMockReducer';

export const renderWithGlobalContext = (
  children: ReactElement,
  deps: Deps = getMockDeps(),
  initState: IGlobalState = getInitialState(),
  combinedReducers: IReducer<IGlobalState, any> = getMockReducer()
) => {
  const renderAPI = render(
    <GlobalProvider deps={deps} initState={initState} combinedReducers={combinedReducers}>
      {children}
    </GlobalProvider>
  );
  return { ...renderAPI, mockReducer: combinedReducers, deps };
};
