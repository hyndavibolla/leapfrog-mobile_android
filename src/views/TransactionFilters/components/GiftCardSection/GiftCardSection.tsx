import React, { memo, useCallback, useContext, useEffect, useMemo } from 'react';
import { FlatList, Pressable, Text } from 'react-native';
import { useIsFocused, useNavigation } from '@react-navigation/native';

import { BubbleSkeleton } from '../BubbleSkeleton';
import { BrandLogo } from '_components/BrandLogo';
import { Carrousel } from '_components/Carrousel';

import { GlobalContext } from '_state_mgmt/GlobalState';
import { useRewardConfigBrandSearch } from '_state_mgmt/reward/hooks';
import { RewardModel } from '_models';
import { TransactionFilter } from '_models/offer';

import { useTestingHelper } from '_utils/useTestingHelper';
import { getNote } from '_utils/transactionFiltersUtil';
import { ENV, ROUTES } from '_constants';
import { numberOfBubbles, numberOfItemsToShow } from '_constants/transactionFilter';

import FallbackIcon from '_assets/fallbackTransactions/fallback.svg';

import { styles } from './styles';

export interface Props {
  transactionType: TransactionFilter;
}

function GiftCardSection({ transactionType }: Props) {
  const { navigate } = useNavigation();
  const isFocused = useIsFocused();
  const { getTestIdProps } = useTestingHelper('gift-card-section');

  const {
    state: {
      reward: {
        config: { categories },
        slideObjectMapByType
      }
    }
  } = useContext(GlobalContext);

  const [onRewardConfigBrandSearch, isLoadingReward = true, rewardError] = useRewardConfigBrandSearch();

  useEffect(() => {
    if (!isFocused) return;
    onRewardConfigBrandSearch();
  }, [isFocused, onRewardConfigBrandSearch]);

  const giftCards = useMemo(() => {
    const configCategoryMap = categories.reduce((total, { id, brands }) => ({ ...total, [id]: brands }), {});

    return configCategoryMap[ENV.CUSTOM_BRAND_CATEGORY_KEY]
      ?.map(({ brandId }) => slideObjectMapByType[RewardModel.SlideObjectType.BRAND][brandId])
      ?.filter(Boolean)
      ?.slice(0, numberOfItemsToShow);
  }, [categories, slideObjectMapByType]);

  const renderGiftCard = useCallback(
    ({ item: { id: brandId, iconUrl } }) => {
      return (
        <Pressable onPress={() => navigate(ROUTES.GIFT_CARD_DETAIL, { brandId })} {...getTestIdProps('gift-card')}>
          <BrandLogo style={[styles.bubble, giftCards?.length <= numberOfBubbles && styles.SpecialBubble]} image={iconUrl} size={60} Fallback={FallbackIcon} />
        </Pressable>
      );
    },
    [getTestIdProps, giftCards?.length, navigate]
  );

  if (isLoadingReward) {
    return <BubbleSkeleton />;
  }

  if (rewardError || !giftCards?.length) {
    return null;
  }

  return (
    <>
      <Text style={styles.note} {...getTestIdProps('note')}>
        {getNote(transactionType)}
      </Text>
      <Carrousel itemWidth={60} separatorWidth={16}>
        <FlatList
          style={[styles.bubbles, giftCards.length <= numberOfBubbles && styles.specialBubbles]}
          contentContainerStyle={[styles.container, giftCards.length <= numberOfBubbles && styles.bubblesContainer]}
          data={giftCards}
          renderItem={renderGiftCard}
          horizontal
          showsHorizontalScrollIndicator={false}
        />
      </Carrousel>
    </>
  );
}

export default memo(GiftCardSection);
