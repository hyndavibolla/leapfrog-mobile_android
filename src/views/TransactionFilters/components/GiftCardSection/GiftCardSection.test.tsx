import React from 'react';
import { act, fireEvent } from '@testing-library/react-native';
import navigation from '@react-navigation/native';

import GiftCardSection, { Props } from './GiftCardSection';

import { getInitialState } from '_state_mgmt/GlobalState';
import { Deps, IGlobalState } from '_models/general';
import { TransactionFilter } from '_models/offer';

import { getSlideBrand_1 } from '_test_utils/entities';
import { getMockDeps } from '_test_utils/getMockDeps';
import { renderWithGlobalContext } from '_test_utils/renderWithGlobalContext';

import { wait } from '_utils/wait';
import { ROUTES } from '_constants';

const mockedNavigate = jest.fn();
jest.mock('@react-navigation/native', () => {
  return { useNavigation: () => ({ navigate: mockedNavigate }) };
});

describe('Gift Card Section', () => {
  let props: Props;
  let deps: Deps;
  let initialState: IGlobalState;

  beforeEach(() => {
    navigation.useIsFocused = jest.fn(() => true);
    props = {
      transactionType: TransactionFilter.ALL_TRANSACTIONS
    };
    deps = getMockDeps();
    initialState = getInitialState();
  });

  it('should render', async () => {
    const { toJSON } = renderWithGlobalContext(<GiftCardSection {...props} />, deps, initialState);
    await act(() => wait(0));
    expect(toJSON()).toMatchSnapshot();
  });

  it('should navigate to gift card detail when the first gift card of the list is pressed', async () => {
    const { getAllByTestId } = renderWithGlobalContext(<GiftCardSection {...props} />, deps, initialState);
    await act(() => wait(0));
    fireEvent.press(getAllByTestId('gift-card-section-gift-card')[0]);
    await act(() => wait(0));
    expect(mockedNavigate).toHaveBeenCalledWith(ROUTES.GIFT_CARD_DETAIL, { brandId: getSlideBrand_1().id });
  });

  it('should render nothing when giftCards list is empty', async () => {
    (deps.raiseService.fetchSlideBrand as jest.Mock).mockRejectedValueOnce('error');
    const { queryAllByTestId } = renderWithGlobalContext(<GiftCardSection {...props} />, deps, initialState);
    await act(() => wait(0));
    expect(queryAllByTestId('gift-card-section-gift-card')[0]).toBeFalsy();
  });

  it('should not fetch gift cards when the component is not focused', async () => {
    navigation.useIsFocused = jest.fn(() => false);
    (deps.raiseService.fetchSlideBrand as jest.Mock).mockRejectedValueOnce('error');
    renderWithGlobalContext(<GiftCardSection {...props} />, deps, initialState);
    await act(() => wait(0));
    expect(deps.raiseService.fetchSlideBrand).not.toBeCalled();
  });
});
