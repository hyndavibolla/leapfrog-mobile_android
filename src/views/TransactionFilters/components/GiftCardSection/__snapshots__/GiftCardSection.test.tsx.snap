// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift Card Section should render 1`] = `
Array [
  <Text
    accessibilityLabel="gift-card-section-note"
    accessible={true}
    style={
      Object {
        "color": "#262C2F",
        "fontFamily": "SFProDisplay-Bold",
        "fontSize": 16,
        "lineHeight": 24,
        "marginBottom": 16,
      }
    }
    testID="gift-card-section-note"
  >
    Most users love these
  </Text>,
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    collapsable={false}
    contentContainerStyle={
      Array [
        Object {
          "alignItems": "center",
        },
        Array [
          Object {
            "paddingBottom": 24,
            "paddingLeft": 24,
            "paddingRight": 8,
            "paddingTop": 24,
          },
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "space-around",
          },
        ],
      ]
    }
    data={
      Array [
        Object {
          "brandName": "Burgers For Days",
          "cardValueConfig": Object {
            "increment": 1,
            "maxValue": 50000,
            "minValue": 500,
            "variableLoadSupported": true,
          },
          "categories": Array [
            Object {
              "name": "Entertainment",
            },
            Object {
              "name": "Travel",
            },
            Object {
              "name": "Automotive",
            },
          ],
          "description": "Buy stuff, buy lots of it.",
          "iconUrl": "https://s3.amazonaws.com/raise-content/ibi/GameStop-Logo.png",
          "id": "754338d5-7c5e-4785-bd05-67cd46545df9",
          "legalTerms": "This card can be used to purchase merchandise only in stores",
          "redemptionConfigs": Object {
            "disclaimer": "Can only use 2 cards",
            "kind": "IN_STORE_OR_ONLINE_ONLY",
            "methods": Array [
              Object {
                "info": "Enter the gift card info online",
                "kind": "ENTER_ONLINE",
              },
              Object {
                "info": "Show the barcode to the cashier",
                "kind": "SHOW_BARCODE",
              },
            ],
          },
          "sortedPosition": 2,
        },
        Object {
          "brandName": "Durum For Weeks",
          "cardValueConfig": Object {
            "increment": 1,
            "maxValue": 50000,
            "minValue": 500,
            "variableLoadSupported": true,
          },
          "categories": Array [
            Object {
              "name": "Entertainment",
            },
          ],
          "description": "Buy stuff, buy lots of it.",
          "iconUrl": "https://s3.amazonaws.com/raise-content/ibi/GameStop-Logo.png",
          "id": "67cd46545df9-754338d5-7c5e-4785-bd05",
          "legalTerms": "This card can be used to purchase merchandise only in stores",
          "redemptionConfigs": Object {
            "disclaimer": "Can only use 2 cards",
            "kind": "IN_STORE_OR_ONLINE_ONLY",
            "methods": Array [
              Object {
                "info": "Enter the gift card info online",
                "kind": "ENTER_ONLINE",
              },
              Object {
                "info": "Show the barcode to the cashier",
                "kind": "SHOW_BARCODE",
              },
            ],
          },
          "sortedPosition": 2,
        },
      ]
    }
    decelerationRate={0}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    showsHorizontalScrollIndicator={false}
    snapToInterval={76}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "flexDirection": "row",
        "marginHorizontal": 0,
        "paddingHorizontal": 24,
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
              },
              Object {
                "width": 60,
              },
            ]
          }
        >
          <View
            accessibilityLabel="gift-card-section-gift-card"
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            testID="gift-card-section-gift-card"
          >
            <View
              style={
                Array [
                  Object {
                    "height": 60,
                    "width": 60,
                  },
                  Array [
                    Object {
                      "marginRight": 16,
                    },
                    Object {
                      "marginRight": 0,
                    },
                  ],
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "alignItems": "center",
                      "borderBottomLeftRadius": 60,
                      "borderBottomRightRadius": 60,
                      "borderTopLeftRadius": 60,
                      "borderTopRightRadius": 60,
                      "height": "100%",
                      "justifyContent": "center",
                      "width": "100%",
                    },
                  ]
                }
              >
                <FastImageView
                  accessibilityLabel="brand-logo-image"
                  accessible={true}
                  onFastImageError={[Function]}
                  onFastImageLoad={[Function]}
                  onFastImageLoadEnd={[Function]}
                  onFastImageLoadStart={[Function]}
                  resizeMode="cover"
                  source={
                    Object {
                      "priority": "high",
                      "uri": "https://s3.amazonaws.com/raise-content/ibi/GameStop-Logo.png",
                    }
                  }
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                  testID="brand-logo-image"
                />
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "minHeight": 1,
              "width": 16,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
              },
              Object {
                "width": 60,
              },
            ]
          }
        >
          <View
            accessibilityLabel="gift-card-section-gift-card"
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            testID="gift-card-section-gift-card"
          >
            <View
              style={
                Array [
                  Object {
                    "height": 60,
                    "width": 60,
                  },
                  Array [
                    Object {
                      "marginRight": 16,
                    },
                    Object {
                      "marginRight": 0,
                    },
                  ],
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "alignItems": "center",
                      "borderBottomLeftRadius": 60,
                      "borderBottomRightRadius": 60,
                      "borderTopLeftRadius": 60,
                      "borderTopRightRadius": 60,
                      "height": "100%",
                      "justifyContent": "center",
                      "width": "100%",
                    },
                  ]
                }
              >
                <FastImageView
                  accessibilityLabel="brand-logo-image"
                  accessible={true}
                  onFastImageError={[Function]}
                  onFastImageLoad={[Function]}
                  onFastImageLoadEnd={[Function]}
                  onFastImageLoadStart={[Function]}
                  resizeMode="cover"
                  source={
                    Object {
                      "priority": "high",
                      "uri": "https://s3.amazonaws.com/raise-content/ibi/GameStop-Logo.png",
                    }
                  }
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                  testID="brand-logo-image"
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>,
]
`;
